docs/GUIDE.md

```diff
@@ -0,0 +1,142 @@
+Codec Registration Checker Compiler Plugin does not need additional configuration, but you can change default configurations as explained below:
+- `--disable`
+
+The Plugin is enabled by default in each project / module where `AkkaSerializationHelperPlugin` is enabled. If you want to disable this particular plugin (but want to keep on using other ASH compiler plugins) &mdash; add following setting inside `build.sbt` for selected project / module:<br>
+`Compile / scalacOptions += "-P:codec-registration-checker-plugin:--disable"`<br><br>
+- `custom cache file path`
+
+This plugin uses a helper cache file named `codec-registration-checker-cache.csv`. The file is created under `target/` directory by default, so that it will not bother users. This file is useful if user performs incremental compilation. If you want, you can specify your custom path instead of the default one. To do so, add following configuration to the plugin:<br>
+`Compile / scalacOptions += "-P:codec-registration-checker-plugin:PATH_TO_CUSTOM_DIRECTORY"`<br><br>
+
+
+### Dump Persistence Schema Compiler Plugin
+Dump Persistence Schema Compiler Plugin does not need additional configuration, but you can change default configurations as explained below:
+- `--disable`
+
+The Plugin is enabled by default in each project / module where `AkkaSerializationHelperPlugin` is enabled. If you want to disable this particular plugin (but want to keep on using other ASH compiler plugins) &mdash; add following setting inside `build.sbt` for selected project / module:<br>
+`Compile / scalacOptions += "-P:dump-persistence-schema-plugin:--disable"`<br><br>
+- `--verbose`
+
+Verbose mode enables additional logs from the plugin. These logs contain detailed info about detected persistence schema. This mode is disabled by default. If you want to enable `verbose` mode, add following setting:<br>
+`Compile / scalacOptions += "-P:dump-persistence-schema-plugin:--verbose"`<br><br>
+
+Dump Persistence Schema Compiler Plugin prepares data for the `ashDumpPersistenceSchema` sbt task, which creates the final output for user.<br>
+
+### ashDumpPersistenceSchema sbt task
+The `ashDumpPersistenceSchema` sbt task dumps schema of akka-persistence to a yaml file. This is the only part of ASH that has to be invoked manually (with `sbt ashDumpPersistenceSchema` command). If you want to use this task, Dump Persistence Schema Compiler Plugin must **not** be disabled. Default output file for this task is `s"${name.value}-dump-persistence-schema-${version.value}.yaml"` (`name.value` and `version.value` are resolved by sbt on your project). If you want to have a custom name for this file, add following setting:<br>
```

Comment:
```suggestion
The `ashDumpPersistenceSchema` sbt task dumps schema of akka-persistence to a yaml file. This is the only part of ASH that has to be invoked explicitly as an sbt task (`sbt ashDumpPersistenceSchema`). If you want to use this task, Dump Persistence Schema Compiler Plugin must **not** be disabled. Default output file for this task is `s"${name.value}-dump-persistence-schema-${version.value}.yaml"` (`name.value` and `version.value` are resolved by sbt on your project). If you want to have a custom name for this file, add following setting:<br>
```

Reply:
Also, pls specify where this file will be located by default (top-level target/? per-module target/?)

Reply:
Also, maybe mention `ashDumpPersistenceSchemaOutputDirectoryPath`?